shell/platform/android/io/flutter/plugin/editing/InputConnectionAdaptor.java

```diff
@@ -38,6 +38,16 @@
   private InputMethodManager mImm;
   private final Layout mLayout;
 
+  // Used to store the last-sent values via updateEditingState to the framework.
+  // These are then compared against to prevent redundant messages with the same
+  // data before any valid operations were made to the contents.
+  private int mPreviousSelectionStart;
```

Comment:
Consider moving the selection start/end, composing start/end, and text fields into a class.  `updateEditingState` would construct an instance of the class from the `mEditable` and call its `equals` method to check whether it matches the previous instance.